### PR TITLE
add org.freedesktop.Sdk.Extension.llvm18

### PR DESCRIPTION
**Pull request by Erick555**
Another LLVM major release SDK extension. Requested in https://github.com/flathub/org.freedesktop.Sdk.Extension.llvm17/issues/5

A usual this preserves all commit history from past version with update commit on top.

### Please confirm your submission meets all the criteria

- [x] Please describe your application briefly.
- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [ ] I have [built][build] and tested the submission locally.
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [ ] I am an author/developer/upstream contributor of the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [ ] The domain used for the application ID is controlled by the application developers either directly or through the code hosting (e.g. GitHub, GitLab, SourceForge, etc.). The [application id guidelines][app-id] are followed.
- [ ] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[maint]: https://docs.flathub.org/docs/for-app-authors/maintanance
[submission]: https://docs.flathub.org/docs/for-app-authors/submission
[build]: https://docs.flathub.org/docs/for-app-authors/submission/#before-submission
[app-id]: https://docs.flathub.org/docs/for-app-authors/requirements#application-id

### CODE REVIEW

**Comment by Erick555** [comment]
bot, build org.freedesktop.Sdk.Extension.llvm18

File: org.freedesktop.Sdk.Extension.llvm18.json [hunk 1]
```diff
@@ -0,0 +1,177 @@
+{
+  "id": "org.freedesktop.Sdk.Extension.llvm18",
+  "branch": "23.08",
+  "runtime": "org.freedesktop.Sdk",
+  "build-extension": true,
+  "sdk": "org.freedesktop.Sdk",
+  "runtime-version": "23.08",
+  "separate-locales": false,
+  "appstream-compose": false,
```
Comment (bbhtt) [inline org.freedesktop.Sdk.Extension.llvm18.json:9]: You can drop this and manual compose, it should work out of the box now.

Comment (Erick555) [reply org.freedesktop.Sdk.Extension.llvm18.json:9]: Do you mean entire:
```
    {
      "name": "appdata",
      "buildsystem": "simple",
      "build-commands": [
        "install -Dm0644 -t ${FLATPAK_DEST}/share/metainfo/ ${FLATPAK_ID}.metainfo.xml",
        "appstream-compose --basename=${FLATPAK_ID} --prefix=${FLATPAK_DEST} --origin=flatpak ${FLATPAK_ID}"
      ],
```

can be dropped?

Comment (bbhtt) [reply org.freedesktop.Sdk.Extension.llvm18.json:9]: No keep the install part drop appstream-compose command and appstream-compose: false

Comment (Erick555) [reply org.freedesktop.Sdk.Extension.llvm18.json]: done

**Comment by Erick555** [comment]
bot, build org.freedesktop.Sdk.Extension.llvm18

**Comment by Erick555** [comment]
bot, build org.freedesktop.Sdk.Extension.llvm18

**Pull request closed without merging**